statsmodels/discrete/truncated_model.py

```diff
@@ -0,0 +1,1047 @@
+                full_output=full_output, disp=0, callback=callback,
+                alpha=alpha_p, trim_mode=trim_mode, auto_trim_tol=auto_trim_tol,
+                size_trim_tol=size_trim_tol, qc_tol=qc_tol, **kwargs).params
+        cntfit = super(CountModel, self).fit_regularized(
+                start_params=start_params, method=method, maxiter=maxiter,
+                full_output=full_output, disp=disp, callback=callback,
+                alpha=alpha, trim_mode=trim_mode, auto_trim_tol=auto_trim_tol,
+                size_trim_tol=size_trim_tol, qc_tol=qc_tol, **kwargs)
+
+        if method in ['l1', 'l1_cvxopt_cp']:
+            discretefit = self.result_reg(self, cntfit)
+        else:
+            raise TypeError(
+                    "argument method == %s, which is not handled" % method)
+
+        return self.result_reg_wrapper(discretefit)
+
+    fit_regularized.__doc__ = DiscreteModel.fit_regularized.__doc__
+
+    def hessian(self, params):
+        """
+        Generic Truncated model Hessian matrix of the loglikelihood
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model
+
+        Returns
+        -------
+        hess : ndarray, (k_vars, k_vars)
+            The Hessian, second derivative of loglikelihood function,
+            evaluated at `params`
+
+        Notes
+        -----
+        """
+        return approx_hess(params, self.loglike)  
+
+
+class TruncatedPoisson(GenericTruncated):
+    """
+    Truncated Poisson model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    truncation : int, optional
+        Truncation parameter specify truncation point out of the support
+        of the distribution. pmf(k) = 0 for k <= truncation
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, offset=None, exposure=None,
+                 truncation=0, missing='none', **kwargs):
+        super(TruncatedPoisson, self).__init__(endog, exog, offset=offset,
+                                               exposure=exposure,
+                                               truncation=truncation,
+                                               missing=missing, **kwargs)
+        self.model_main = Poisson(self.endog, self.exog,
+                                  exposure=exposure,
+                                  offset=offset)
+        self.model_dist = truncatedpoisson
+        self.result = TruncatedPoissonResults
```

Comment:
these should have `class` in the name as other count models
it looks like we only have it in ZI models `self.result_class` ,...